GEECS-Scanner-GUI/geecs_scanner/app/action_library.py

```diff
@@ -0,0 +1,510 @@
+                self.ui.lineActionOption2.setText(action.get("variable"))
+                self.ui.labelActionOption3.setText("Set Value:")
+                self.ui.lineActionOption3.setEnabled(True)
+                self.ui.lineActionOption3.setText(str(action.get("value")))
+            elif action['action'] == 'get':
+                self.action_mode = 'get'
+                self.ui.labelActionOption1.setText("GEECS Device Name:")
+                self.ui.lineActionOption1.setEnabled(True)
+                self.ui.lineActionOption1.setText(action.get("device"))
+                self.ui.labelActionOption2.setText("Variable Name:")
+                self.ui.lineActionOption2.setEnabled(True)
+                self.ui.lineActionOption2.setText(action.get("variable"))
+                self.ui.labelActionOption3.setText("Expected Value:")
+                self.ui.lineActionOption3.setEnabled(True)
+                self.ui.lineActionOption3.setText(str(action.get("expected_value")))
+
+    def update_action_info(self):
+        """Updates the backend action dictionary when one of the line edits for options is changed"""
+        index = self.get_selected_step_index()
+        name = self.get_selected_name()
+        if index >= 0 and name:
+            action = self.actions_data['actions'][name]['steps'][index]
+
+            if action.get("wait") is not None:
+                action['wait'] = parse_variable_text(self.ui.lineActionOption1.text().strip())
+            elif action['action'] == 'execute':
+                action['action_name'] = self.ui.lineActionOption1.text().strip()
+            elif action['action'] == 'run':
+                action["file_name"] = self.ui.lineActionOption1.text().strip()
+                action["class_name"] = self.ui.lineActionOption2.text().strip()
+            elif action['action'] == 'set':
+                action["device"] = self.ui.lineActionOption1.text().strip()
+                action["variable"] = self.ui.lineActionOption2.text().strip()
+                action["value"] = parse_variable_text(self.ui.lineActionOption3.text().strip())
+            elif action['action'] == 'get':
+                action["device"] = self.ui.lineActionOption1.text().strip()
+                action["variable"] = self.ui.lineActionOption2.text().strip()
+                action["expected_value"] = parse_variable_text(self.ui.lineActionOption3.text().strip())
+
+            self.update_action_list(index=index)
+
+    def get_selected_step_index(self) -> int:
+        """Finds the location of the currently-selected action in the visible list on the GUI.
+
+        :return: If no selection, returns -1.  Otherwise, return the index of the currently selected step
+        """
+        selected_action = self.ui.listActionSteps.selectedItems()
+        if not selected_action:
+            return -1
+        for action in selected_action:
+            index = self.ui.listActionSteps.row(action)
+            return index
+
```

Comment:
index = self.ui.listActionSteps.row(selected_action[0])
return index
```suggestion
        for action in selected_action:
            index = self.ui.listActionSteps.row(action)
            return index

```